importer/pipeline/pipeline.go

```diff
@@ -0,0 +1,268 @@
+	}
+
+	return &Pipeline{
+		fetcher:     fetcher,
+		transformer: transformer,
+		pusher:      pusher,
+		config:      config,
+	}
+}
+
+// Run executes the pipeline.
+func (p *Pipeline) Run(ctx context.Context) (*Result, error) {
+	result := &Result{}
+
+	// Stage 1: Fetch records
+	fetchedCh, fetchErrCh := p.fetcher.Fetch(ctx)
+
+	// Stage 2: Transform records
+	transformedCh, transformErrCh := p.runTransformStage(ctx, fetchedCh, result)
+
+	// Stage 3: Push records (if not dry-run)
+	var pushErrCh <-chan error
+	if !p.config.DryRun {
+		pushErrCh = p.runPushStage(ctx, transformedCh, result)
+	}
+
+	// Collect errors from all stages
+	var wg sync.WaitGroup
+	wg.Add(3) //nolint:mnd
+
+	// Collect fetch errors
+	go func() {
+		defer wg.Done()
+
+		for err := range fetchErrCh {
+			if err != nil {
+				result.mu.Lock()
+				result.Errors = append(result.Errors, fmt.Errorf("fetch error: %w", err))
+				result.mu.Unlock()
+			}
+		}
+	}()
+
+	// Collect transform errors
+	go func() {
+		defer wg.Done()
+
+		for err := range transformErrCh {
+			if err != nil {
+				result.mu.Lock()
+				result.Errors = append(result.Errors, err)
+				result.mu.Unlock()
+			}
+		}
+	}()
+
+	// Collect push errors
+	go func() {
+		defer wg.Done()
+
+		for err := range pushErrCh {
+			if err != nil {
+				result.mu.Lock()
+				result.Errors = append(result.Errors, err)
+				result.mu.Unlock()
+			}
+		}
+	}()
+
+	wg.Wait()
+
+	return result, nil
+}
+
+// runTransformStage runs the transformation stage with concurrent workers.
+func (p *Pipeline) runTransformStage(ctx context.Context, inputCh <-chan interface{}, result *Result) (<-chan *corev1.Record, <-chan error) {
+	outputCh := make(chan *corev1.Record)
+	errCh := make(chan error)
+
+	var wg sync.WaitGroup
+
+	// Start transformer workers
+	for range p.config.TransformerWorkers {
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case source, ok := <-inputCh:
+					if !ok {
+						return
+					}
+
+					// Track total records
+					result.mu.Lock()
+					result.TotalRecords++
+					result.mu.Unlock()
+
+					// Transform the record
+					record, err := p.transformer.Transform(ctx, source)
+					if err != nil {
+						result.mu.Lock()
+						result.FailedCount++
+						result.mu.Unlock()
+
+						select {
+						case errCh <- fmt.Errorf("transform error: %w", err):
+						case <-ctx.Done():
+							return
+						}
+
+						continue
+					}
+
+					// Send transformed record to next stage
+					select {
+					case outputCh <- record:
+					case <-ctx.Done():
+						return
+					}
+				}
+			}
+		}()
+	}
+
+	// Close output channel when all workers are done
+	go func() {
+		wg.Wait()
+		close(outputCh)
+		close(errCh)
+	}()
+
+	return outputCh, errCh
+}
+
+// runPushStage runs the push stage with concurrent workers.
+func (p *Pipeline) runPushStage(ctx context.Context, inputCh <-chan *corev1.Record, result *Result) <-chan error {
```

Comment:
We can remove this. New Pusher using PushStream and PushStream uses a single bidirectional gRPC stream to pipeline all records (achieving ~8x lower latency per record and 80% less memory) instead of spawning N workers that each make individual RPC calls with full request/response overhead.